File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.js
/**
 * @param {number[]} cardPoints
 * @param {number} k
 * @return {number}
 */
var maxScore = function(cardPoints, k) {
    const n = cardPoints.length;
    let sum = cardPoints.slice(0, k).reduce((a, b) => a + b);
    let ans = sum;
    
    for (let i = 1; i <= k; i++) {
        sum += cardPoints[n - i] - cardPoints[k - i];
        ans = Math.max(ans, sum);
    }
    
    return ans;
};

File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.py
class Solution:
    def maxScore(self, cardPoints: List[int], k: int) -> int:
        s = sum(cardPoints[:k])
        ans, n = s, len(cardPoints)
        for i in range(1, k + 1):
            s += cardPoints[n - i] - cardPoints[k - i]
            ans = max(ans, s)
        return ans